src/lib/features/client-feature-toggles/client-feature-toggle.controller.ts

```diff
@@ -357,11 +357,11 @@ export default class FeatureController extends Controller {
     }
 
     async calculateMeta(query: IFeatureToggleQuery): Promise<IMeta> {
-        // TODO: We will need to standardize this to be able to implement this a cross languages (Edge in Rust?).
         const revisionId =
-            await this.configurationRevisionService.getMaxRevisionId();
+            await this.configurationRevisionService.getMaxRevisionId(
+                query.environment,
```

Comment:
We should now consider the environment if calculating the etag for an environment specific token.